MITADS-Speech/corpora_importer.py

```diff
@@ -141,32 +151,26 @@ def _maybe_convert_sets(self,corpus:Corpus):
         if(num_samples==0):
             return
 
-        if(corpus.make_wav_resample):
-                        
-            # Mutable counters for the concurrent embedded routine
-            counter = get_counter()
-            print(f"Converting wav/mp3 files to wav {SAMPLE_RATE}hz...")
-            pool = Pool()
-            bar = progressbar.ProgressBar(max_value=num_samples, widgets=SIMPLE_BAR)
-            rows = []
-            for i, processed in enumerate(pool.imap_unordered(self.one_sample,samples), start=1):
-                counter += processed[0]
-                rows += processed[1]
-                bar.update(i)
-            bar.update(num_samples)
-            pool.close()
-            pool.join()
-            
-            ## rows contains wav filenames filtered
-            filenames_filtered = [r[0] for r in rows]
-            for f in corpus.audios:
-                if f not in filenames_filtered:
-                    ##remove item
-                    corpus.audios.remove(f)
-                    del corpus.utterences[f]
-            ########################################
-
-        self._write_csv(corpus)
+        ## all examples are processed, even if the resample is not necessary, the duration or other filters should be evaluated
+        samples = [ [a,corpus.make_wav_resample] for a in corpus.audios ] 
+        ##self.one_sample(samples[0])
+        # Mutable counters for the concurrent embedded routine
+        counter = get_counter()
+        print(f"Converting wav/mp3 files to wav {SAMPLE_RATE}hz...")
+        pool = Pool()
+        bar = progressbar.ProgressBar(max_value=num_samples, widgets=SIMPLE_BAR)
+        rows = []
+        for i, processed in enumerate(pool.imap_unordered(self.one_sample,samples), start=1):
+            counter += processed[0]
+            rows += processed[1]
+            bar.update(i)
+        bar.update(num_samples)
+        pool.close()
+        pool.join()        
+       
+        ########################################
+        ## filtered rows data are evaluated in write_csv
+        self._write_csv(corpus,rows)
 
     def _maybe_convert_wav(self,mp3_filename, wav_filename):
```

Comment:
in this function there is a typo: CHANNELS must be N_CHANNELS

Reply:
yes, i fix!